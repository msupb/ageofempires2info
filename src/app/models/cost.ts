export interface ICost {
    info?: string;
    Wood?: number;
    Food?: number;
    Stone?: number;
    Gold?: number;
}