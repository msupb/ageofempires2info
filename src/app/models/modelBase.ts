export interface IModelBase {
    localId?: string;
    category?: string;
    id: number;
    name: string;
}