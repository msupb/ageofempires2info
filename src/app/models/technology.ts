import { ICost } from "./cost";

export interface ITechnology {
    id: number;
    name: string;
    description: string;
    expansion: string;
    age: string;
    develops_in: string;
    cost: ICost;
    build_time: number;
    applies_to: Array<string>;
}