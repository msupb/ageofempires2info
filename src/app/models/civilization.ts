import {IModelBase} from './modelBase';

export interface ICivilization extends IModelBase {
    expansion: string;
    army_type: string;
    unique_unit: Array<string>;
    unique_tech: Array<string>;
    team_bonus: string;
    civilization_bonus: Array<string>;
}