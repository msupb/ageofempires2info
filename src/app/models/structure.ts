import { ICost } from "./cost";
import {IModelBase} from './modelBase';

export interface IStructure extends IModelBase {
    description: string;
    expansion: string;
    age: string;
    cost: ICost;
    build_time: number;
    hit_points: number;
    line_of_sight: number;
    armor: string;
    range: string;
    relaod_time: number;
    attack: number;
    special: Array<string>
}