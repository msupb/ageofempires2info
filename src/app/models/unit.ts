import { ICost } from "./cost";
import {IModelBase} from './modelBase';

export interface IUnit extends IModelBase {
    description: string;
    expansion: string;
    age: string;
    created_in: string;
    cost: ICost;
    build_time: number;
    reload_time: number;
    attack_delay: number;
    movement_rate: number;
    line_of_sight: number;
    hit_points: number;
    range: string;
    attack: number;
    armor: string;
    attack_bonus: Array<string>;
    armor_bonus: Array<string>;
    search_radius: number;
    accuracy: string;
    blast_radius: number;
}