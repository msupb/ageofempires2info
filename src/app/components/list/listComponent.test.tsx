import React from 'react';
import { cleanup, render, within } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import ListComponent from './listComponent';
import { IModelBase } from '../../models/modelBase';
import ListItem from './listItem';
import TestRenderer from 'react-test-renderer'

const testItem = {
    'localId': 'testid',
    'category': 'testCategory',
    'id': 0,
    'name': 'testName',
    'age': 'testAge'
    }

const testList: Array<IModelBase> = [testItem];

afterEach(cleanup);

test("ListComponent contains one <ListItem></ListItem>", () => {
     const { getAllByTestId, getByTestId } = render(<Router><ListComponent itemList={testList}></ListComponent></Router>);
     const listComponent = getByTestId('listComponent');
     const listItemInListComponent = within(listComponent).getAllByTestId('listItem');
     expect(listItemInListComponent.length).toBe(1);
})

it("matches ListComponent snapshot", () => {
    const jsonObj = TestRenderer.create(<Router><ListComponent itemList={testList}></ListComponent></Router>).toJSON();
    expect(jsonObj).toMatchSnapshot();
})

test("ListItem contains one <CusButton></CusButton>", () => {
    const { getAllByTestId, getByTestId } = render(<Router><ListItem className='list-card card' key="testkey" item={testItem}></ListItem></Router>);
    const listItem = getByTestId('listItem');
    const cusButtonInListItem = within(listItem).getAllByTestId('cusButtonTest');
    expect(cusButtonInListItem.length).toBe(1);
})

it("matches ListItem snapshot", () => {
    const jsonObj = TestRenderer.create(<Router><ListItem className='list-card card' key="testkey" item={testItem}></ListItem></Router>).toJSON();
    expect(jsonObj).toMatchSnapshot();
})


