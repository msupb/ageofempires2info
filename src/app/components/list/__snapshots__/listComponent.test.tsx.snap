// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches ListComponent snapshot 1`] = `
<div
  className="card"
  data-testid="listComponent"
>
  <div
    className="container"
  >
    <div
      className="list-card card"
      data-testid="listItem"
    >
      <div
        className="container"
      >
        <h4>
          <b>
            testName
          </b>
        </h4>
        <span
          className="button-span"
        >
          <a
            href="/details/testCategory/0"
            onClick={[Function]}
          >
            <button
              data-testid="cusButtonTest"
              onClick={[Function]}
              type="button"
            >
              Details
            </button>
          </a>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`matches ListItem snapshot 1`] = `
<div
  className="list-card card"
  data-testid="listItem"
>
  <div
    className="container"
  >
    <h4>
      <b>
        testName
      </b>
    </h4>
    <span
      className="button-span"
    >
      <a
        href="/details/testCategory/0"
        onClick={[Function]}
      >
        <button
          data-testid="cusButtonTest"
          onClick={[Function]}
          type="button"
        >
          Details
        </button>
      </a>
    </span>
  </div>
</div>
`;
